pyseir/cli.py

```diff
@@ -137,51 +137,138 @@ def _map_outputs(
             )
 
 
-def _run_all(
-    state=None,
+def _state_only_pipeline(
+    state,
+    states_only=False,
     run_mode=DEFAULT_RUN_MODE,
-    generate_reports=False,
+    generate_reports=True,
     output_interval_days=1,
+    output_dir=None,
+):
+    _infer_rt(state, states_only=states_only)
+    _run_mle_fits(state, states_only=states_only)
+    _run_ensembles(
+        state,
+        ensemble_kwargs=dict(
+            run_mode=run_mode,
+            generate_report=generate_reports,
+            covid_timeseries=nyt_dataset,
+        ),
+        states_only=states_only,
+    )
+    if generate_reports:
+        _generate_state_reports(state)
+    # remove outputs atm. just output at the end
+    # _map_outputs(
+    #     state,
+    #     output_interval_days,
+    #     states_only=states_only,
+    #     output_dir=output_dir,
+    #     run_mode=run_mode,
+    # )
+
+
+def _build_all_for_states(
+    states=[],
+    run_mode=DEFAULT_RUN_MODE,
+    generate_reports=True,
+    output_interval_days=4,
     skip_download=False,
-    states_only=False,
     output_dir=None,
     skip_whitelist=False,
 ):
-
+    # prepare data
     _cache_global_datasets()
-
     if not skip_download:
         cache_all_data()
-
     if not skip_whitelist:
         _generate_whitelist()
 
+    # run states in paralell
+    all_county_fips = []
+    for state in states:
+        all_county_fips += model_fitter.build_county_list(state)
+
+    # do everything for just states in paralell
+    states_only_func = partial(
+        _state_only_pipeline,
+        run_mode=run_mode,
+        generate_reports=generate_reports,
+        output_interval_days=output_interval_days,
+        states_only=True,
+        output_dir=output_dir,
+    )
+    p = Pool()
```

Comment:
Can just use a single pool and p.join() between steps until the very end where a single `p.close()` will work.